code/game/gamemodes/shadowling/shadowling_sight.dm

```diff
@@ -0,0 +1,112 @@
+// Based off of /datum/camerachunk, but for shadowlings
+
+/datum/shadowchunk
+	var/list/overlays = list()
+	var/list/turfs = list()
+	var/list/seenby = list()
+	var/visible = 0
+	var/changed = 0
+	var/updating = 0
+	var/x = 0
+	var/y = 0
+	var/z = 0
+
+/mob/var/list/visible_shadow_chunks = list()
+
+// Add a shadowling to the chunk, then update if changed.
+
+/datum/shadowchunk/proc/add(mob/S)
+	if(!S)
+		return
+	S.visible_shadow_chunks += src
+	if(S.client)
+		S.client.images += overlays
+	visible++
+	seenby += S
+	//if(changed && !updating)
+	//	update()
+
+// Remove a shadowling from the chunk, then update if changed.
+
+/datum/shadowchunk/proc/remove(mob/S)
+	if(!S)
+		return
+	S.visible_shadow_chunks -= src
+	if(S.client)
+		S.client.images -= overlays
+	seenby -= S
+	if(visible > 0)
+		visible--
+
+/turf/var/image/shadowling_bright_overlay
+/turf/var/atom/movable/bright_overlay_container
+
+/datum/shadowchunk/New(loc, x, y, z)
+
+	// 0xf = 15
+	x &= ~0xf
```

Comment:
Don't use hex. It's harder to read, and all of our code uses directly defined numbers.